### DISCUSSION TITLE
Create Auth Request via API

### DISCUSSION THREAD

**Discussion by z3ttee** [zitadel/zitadel#8227] [Q&A]
Hey all,
I'm currently part of a team that implements a custom login UI for Zitadel.
We need to be OIDC compliant and therefor can't work with sessions alone. For me its not clear how to get an auth request id without calling the authorize URL of Zitadel. Its a bit hacky to rely on redirects, instead of calling an API to create a new auth request.

Maybe I'm overseeing something here. But wouldn't it be better to have an endpoint so devs can create auth requests by calling that endpoint? 

Hope someone can help me here :) or knows what the current state is, because in the docs I can't find a clear answer to this.

**Comment by hifabienne** [comment]
Hi @z3ttee 
Great to hear that you implement a custom ui.
Not sure if you already have looked at this docs page: https://zitadel.com/docs/guides/integrate/login-ui/oidc-standard
So you will do an auhorize request but it will be to your custom login and that will proxy the request to the ZITADEL backend.
You should then be redirected back, and have the authrequest included
It is described in this section: https://zitadel.com/docs/guides/integrate/login-ui/oidc-standard#authorize-request

**Reply by z3ttee** [reply]
Hi @hifabienne,
yeah I already looked at the linked docs. But I'm currently running into the issue, where I'm not correctly redirected from the `${ZITADEL_URL}/oauth/v2/authorize` endpoint. Instead of redirecting to `${DOMAIN_OF_CUSTOM_UI}/login?authRequestId=...` I get redirected to `${ZITADEL_URL}/login?authRequest`.

I am currently "proxying" the authorize request by sending data to an api and send a request to Zitadel from that api. I know that is not the right way to proxy the request, but this is the only way I know to enable local development. Using a proxy on local machines can become a painpoint for us.

Using this approach, the creation of a new auth request works fine, but its a bit hacky I think.
Currently there is an OIDCService mentioned in the zitadel api docs. Using this service its already possible to get auth request details and finalize a request, but we can't create an auth request there.
That's why I asked if its possible to have a direct call to the zitadel api to create a new auth request.?

**Reply by hifabienne** [reply]
@peintnermax or @livio-a can you help here?

**Comment by reactima** [comment]
Read Support for the OpenID Connect(OIDC) Standard in a Custom Login UI
https://zitadel.com/docs/guides/integrate/login-ui/oidc-standard

zitadel/typescript repo seems missing /authorize endpoint

so you have to go through ${ZITADEL_URL}/oauth/v2/authorize endpoint

getting authRequestId is puzzling

**Reply by fforootd** [reply]
Yeah our docs are not yet there ;-)

Btw. I just replied to https://github.com/zitadel/typescript/issues/188
